x/wasm/keeper/query_plugins_test.go

```diff
@@ -356,6 +357,94 @@ func TestBankQuerierBalance(t *testing.T) {
 	assert.Equal(t, exp, got)
 }
 
+func TestBankQuerierMetadata(t *testing.T) {
+	metadata := banktypes.Metadata{
+		Name: "Test Token",
+		Base: "utest",
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    "utest",
+				Exponent: 0,
+			},
+		},
+	}
+
+	mock := bankKeeperMock{GetDenomMetadataFn: func(ctx sdk.Context, denom string) (banktypes.Metadata, bool) {
+		if denom == "utest" {
+			return metadata, true
+		} else {
+			return banktypes.Metadata{}, false
+		}
+	}}
+
+	ctx := sdk.Context{}
+	q := keeper.BankQuerier(mock)
+	gotBz, gotErr := q(ctx, &wasmvmtypes.BankQuery{
+		DenomMetadata: &wasmvmtypes.DenomMetadataQuery{
+			Denom: "utest",
+		},
+	})
+	require.NoError(t, gotErr)
+	var got wasmvmtypes.DenomMetadataResponse
+	require.NoError(t, json.Unmarshal(gotBz, &got))
+	exp := wasmvmtypes.DenomMetadata{
+		Name: "Test Token",
+		Base: "utest",
+		DenomUnits: []*wasmvmtypes.DenomUnit{
+			{
+				Denom:    "utest",
+				Exponent: 0,
+			},
+		},
+	}
+	assert.Equal(t, exp, got.Metadata)
+}
+
+func TestBankQuerierAllMetadata(t *testing.T) {
+	metadata := []banktypes.Metadata{
+		{
+			Name: "Test Token",
+			Base: "utest",
+			DenomUnits: []*banktypes.DenomUnit{
+				{
+					Denom:    "utest",
+					Exponent: 0,
+				},
+			},
+		},
+	}
+
+	mock := bankKeeperMock{GetDenomsMetadataFn: func(ctx context.Context, req *banktypes.QueryDenomsMetadataRequest) (*banktypes.QueryDenomsMetadataResponse, error) {
+		return &banktypes.QueryDenomsMetadataResponse{
+			Metadatas: metadata,
+		}, nil
+	}}
+
+	ctx := sdk.Context{}
+	q := keeper.BankQuerier(mock)
+	gotBz, gotErr := q(ctx, &wasmvmtypes.BankQuery{
+		AllDenomMetadata: &wasmvmtypes.AllDenomMetadataQuery{},
+	})
+	require.NoError(t, gotErr)
+	var got wasmvmtypes.AllDenomMetadataResponse
+	require.NoError(t, json.Unmarshal(gotBz, &got))
+	exp := wasmvmtypes.AllDenomMetadataResponse{
+		Metadata: []wasmvmtypes.DenomMetadata{
+			{
+				Name: "Test Token",
+				Base: "utest",
+				DenomUnits: []*wasmvmtypes.DenomUnit{
+					{
+						Denom:    "utest",
+						Exponent: 0,
+					},
+				},
+			},
+		},
+	}
+	assert.Equal(t, exp, got)
+}
```

Comment:
Thanks for writing tests! For legacy reasons, the query plugin has coverage not as high as other parts. Therefore I would love to improve this as we go. I don't want to push all the work on you but in case you are full into testing, it would be good to cover the unknown denom case as well. 
The pagination data from the wasmvm request is also not covered. It would be good to have a test case with and without pagination set. You could capture the value in the bank mock and compare it to match the expectations later, for example

Reply:
Most of the tests were written by [nik-suri](https://github.com/nik-suri) (the original contributor). I added the unknown denom case now and one test for comparing the supplied pagination with the converted one (I hope that's what you meant).